services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +44,57 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
+  if (thermalMgrQueueHandle == NULL){
+    return ERR_CODE_INVALID_STATE;
+  }
+  else if(event == NULL){
+    return ERR_CODE_INVALID_ARG;
+  }
+  else if (xQueueSend(thermalMgrQueueHandle, (void *) event, (TickType_t) 0) == errQUEUE_FULL){
+    return ERR_CODE_QUEUE_FULL;
+  }
+
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = THERMAL_MGR_EVENT_INTERRUPT;
+  thermalMgrSendEvent(&interrupt);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  float *temp;
```

Comment:
this shouldnt be declared as an ununitialized pointer since its just NULL right now so it isn't pointing to any memory where u can actually store some temp data. It should just be `float temp;`

Reply:
also in general try to limit the scope of variables as small as possible so movethis into the while loop